### PR TITLE
Make `ssc build` fail early when `[meta] build_identifier` is not provided

### PR DESCRIPTION
**Pull request by chicoxyzzy**
TODO:
- [x] check that `build_identifier` is a valid string
- [x] check for other required fields (checks are there already)

Bonus:
- [x] more config docs

### CODE REVIEW

File: src/cli/cli.cc [hunk 1]
```diff
@@ -2415,6 +2411,11 @@ int main (const int argc, const char* argv[]) {
   // Insert the elements of runOptions into buildOptions
   buildOptions.insert(buildOptions.end(), runOptions.begin(), runOptions.end());
   createSubcommand("build", buildOptions, true, [&](Map optionsWithValue, std::unordered_set<String> optionsWithoutValue) -> void {
+    if (settings["meta_bundle_identifier"].size() == 0) {
```
Comment (jwerle) [inline src/cli/cli.cc:2414]: we should validate that the build is not for an extension

File: src/cli/templates.hh [hunk 2]
```diff
@@ -1443,6 +1452,9 @@ maintainer = "Beep Boop Corp."
 ; The title of the app used in metadata files. This is NOT a window title. Can contain spaces and special characters. Defaults to name in a [build] section.
 title = "Beep Boop"
 
+; Builds an extension when set to "extension".
+type = "extension"
```
Comment (jwerle) [inline src/cli/templates.hh:1456]: should this be here?

Comment (chicoxyzzy) [reply src/cli/templates.hh:1456]: Do you think we should comment it or it's better to remove it completely?

Comment (jwerle) [reply src/cli/templates.hh:1456]: Comment it out and set it to an empty string (default)

**Review by jwerle — APPROVED** [review]

**Pull request merged**